src/devices/src/virtio/block/io/mod.rs

```diff
@@ -198,32 +198,22 @@ pub mod tests {
         };
     }
 
-    macro_rules! assert_async_execution {
-        ($engine:expr, $count:expr) => {
-            if let FileEngine::Async(ref mut engine) = $engine {
-                engine.kick_submission_queue_and_wait(1).unwrap();
-                assert_eq!(engine.pop().unwrap().unwrap().result().unwrap(), $count);
-            }
-        };
+    fn assert_async_execution(engine: &mut FileEngine<()>, count: u32) {
+        if let FileEngine::Async(ref mut engine) = engine {
+            engine.kick_submission_queue_and_wait(1).unwrap();
+            assert_eq!(engine.pop().unwrap().unwrap().result().unwrap(), count);
+        }
```

Comment:
I don't really understand why this change is required. Could you please provide more detail?